R/variable.R

```diff
@@ -211,8 +211,17 @@ FactorToNumeric <- function(x, binary = TRUE, name = NULL, remove.first = TRUE)
 FactorToIndicators <- function(variable, name = NULL)
 {
     if (is.null(name))
-        name = Names(variable)
-    result <- stats::model.matrix( ~ variable - 1)
+        name <- Names(variable)
+    if (length(levels(variable)) == 1) { 
```

Comment:
can use `nlevels` for this instead of `length(levels(x))` for readability.

Reply:
agreed